### ISSUE TITLE
Smart completion - Support editing properties files

### ISSUE THREAD

**Issue by davsclaus** [davsclaus/camel-idea-plugin#13]
If you are editing Camel endpoints in properties file, when you are using property placeholders. 

So its not only Java but also .properties files as well (text files).

**Comment by fharms** [comment]
Could you give an example what you mean by smart completion in properties files?

**Comment by davsclaus** [comment]
Oh for example you may externalize your Camel endpoint uris to plain .properties file and have a Camel route in Java with:

    from("{{foo}}")
     .to("bean:foo");

And then have a properties file myproperties.properties

   foo=jms:queue:bar?concurrentConsumers=5

Then you want IDEA to offer the same Camel endpoint smart completions in the properties file, as if you edit the endpoint in the Java code (or XML). So if you put the cursor at

   foo=jms:queue:bar?concurrentConsumers=5<CURSOR HERE>

then you can press ctrl + space to see a list of options you can use etc.

**Comment by fharms** [comment]
You can assign me to this, if you need help

**Comment by davsclaus** [comment]
Yeah try to set language = any for that documentation provider to see if it also works for property files as well.

**Comment by davsclaus** [comment]
I pushed some polish. Such as import order and renamed a class so it had a better name, and removed some of the old deprecated stuff.

**Comment by davsclaus** [comment]
Great work @fharms we have nice smart completion in properties files now
